src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.SignVerify.cs

```diff
@@ -0,0 +1,179 @@
+                    dataHash.Length,
+                    pSignature,
+                    signature.Length,
+                    hashAlgorithm,
+                    signatureAlgorithm,
+                    out pErrorOut);
+            }
+        }
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static unsafe extern int AppleCryptoNative_SecKeyVerifySignature(
+            SafeSecKeyRefHandle publicKey,
+            byte* pbDataHash,
+            int cbDataHash,
+            byte* pbSignature,
+            int cbSignature,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm,
+            out SafeCFErrorHandle pErrorOut);
+
+        private static unsafe int AppleCryptoNative_SecKeyCreateSignature(
+            SafeSecKeyRefHandle privateKey,
+            ReadOnlySpan<byte> dataHash,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm,
+            out SafeCFDataHandle pSignatureOut,
+            out SafeCFErrorHandle pErrorOut)
+        {
+            fixed (byte* pDataHash = dataHash)
+            {
+                return AppleCryptoNative_SecKeyCreateSignature(
+                    privateKey,
+                    pDataHash,
+                    dataHash.Length,
+                    hashAlgorithm,
+                    signatureAlgorithm,
+                    out pSignatureOut,
+                    out pErrorOut);
+            }
+        }
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static unsafe extern int AppleCryptoNative_SecKeyCreateSignature(
+            SafeSecKeyRefHandle privateKey,
+            byte* pbDataHash,
+            int cbDataHash,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm,
+            out SafeCFDataHandle pSignatureOut,
+            out SafeCFErrorHandle pErrorOut);
+
+        internal static bool VerifySignature(
+            SafeSecKeyRefHandle publicKey,
+            ReadOnlySpan<byte> dataHash,
+            ReadOnlySpan<byte> signature,
+            PAL_HashAlgorithm hashAlgorithm,
+            PAL_SignatureAlgorithm signatureAlgorithm)
+        {
+            const int Valid = 1;
+            const int Invalid = 0;
+
+            int result = AppleCryptoNative_SecKeyVerifySignature(
+                publicKey,
+                dataHash,
+                signature,
+                hashAlgorithm,
+                signatureAlgorithm,
+                out SafeCFErrorHandle errorHandle);
+
+            using (errorHandle)
+            {
+                return result switch
+                {
+                    Valid => true,
+                    Invalid => false,
+                    kErrorSeeError => throw CreateExceptionForCFError(errorHandle),
+                    kPlatformNotSupported => throw new PlatformNotSupportedException(),
+                    _ => throw new CryptographicException { HResult = result }
```

Comment:
This is unexpected, right?  So a Debug.Fail and a blank CryptographicException (e.g. don't set HResult).

That probably requires changing expression switch to statement switch.